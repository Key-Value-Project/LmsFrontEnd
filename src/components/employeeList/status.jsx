import React from 'react';

export const Status = ({ status }) => {
    const statusColor = () => {
        switch (status) {
            case 'Active':
                return '#c6f5b0';
            case 'In Active':
                return '#f2bdc1';
            case 'On Leave':
                return '#f5eba9';
            default:
                return '#f5f5f5';
        }
    };

    const statusStyle = {
        width: '50%',
        minWidth: 'fit-content',
        backgroundColor: statusColor(),
        borderRadius: '15px',
        padding: '5px 15px',
        cursor: 'pointer',
        textAlign: 'center',
    };

    return (
        <p style={statusStyle} data-testid="test-status">
            {status}
        </p>
    );
};
