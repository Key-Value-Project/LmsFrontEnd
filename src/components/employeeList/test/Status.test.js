import { render } from "@testing-library/react";
import { Status } from "../status";
import React from "react";

describe("Check Status Component", () => { 
    test("Rendered", () => {
        const { getByTestId } = render(<Status />);
        const status = getByTestId("test-status");
        expect(status).toBeTruthy();
    });
    test("Status Displayed", () => {
        const status = "Active";
        const { getByText } = render(<Status status={status} />);
        expect(getByText("Active")).toBeInTheDocument();
    });
    test("Status Color", () => {
        const status = "Active";
        const { getByText } = render(<Status status={status} />);
        const statusElement = getByText("Active");
        expect(statusElement.style.backgroundColor).toBe("rgb(198, 245, 176)");
    });
    test("Status Color", () => {
        const status = "In Active";
        const { getByText } = render(<Status status={status} />);
        const statusElement = getByText("In Active");
        expect(statusElement.style.backgroundColor).toBe("rgb(242, 189, 193)");
    });
    test("Status Color", () => {
        const status = "On Leave";
        const { getByText } = render(<Status status={status} />);
        const statusElement = getByText("On Leave");
        expect(statusElement.style.backgroundColor).toBe("rgb(245, 235, 169)");
    });
    test("Snapshot", () => {
        const status = "Active";
        const { asFragment } = render(<Status status={status} />);
        expect(asFragment()).toMatchSnapshot();
    });
});


