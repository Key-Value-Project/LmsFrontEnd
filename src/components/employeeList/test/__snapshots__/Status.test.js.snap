// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Status Component Snapshot 1`] = `
<DocumentFragment>
  <p
    data-testid="test-status"
    style="background-color: rgb(198, 245, 176); border-radius: 15px; padding: 5px 15px; cursor: pointer;"
  >
    Active
  </p>
</DocumentFragment>
`;
